### ISSUE TITLE
Utility tool to analyse distributions of performance data

### ISSUE THREAD

**Issue by bredamatt** [paritytech/polkadot#6677]
TLDR:
The utility tool would give us the ability to analyse empirically the impact of performance related changes we make going forwards.

Once spans are completed in performance critical subsystems (approval-voting, for example), it is possible to enable "span-metrics" in grafana tempo to get histograms of durations of spans as metrics in prometheus, for example:

![Image](https://user-images.githubusercontent.com/28816406/217023764-9b43f1e1-2e51-4546-a846-64c48cdfc32d.png)

This type of data is useful for doing statistical analysis on, and hence, it would be interesting to leverage this information in the performance analysis of changes made to subsystems accordingly. To do this, it may be valuable with a small utility tool that can do some statistical analysis on the "span-metrics", but also on any other performance data we have available for us.

Example:
This could be in CI, or integrated into Zombienet, or Versi even. Imagine running a small network for 10 minutes where we extract the performance data over this time period as usual. 10 minutes would give us 100 samples of `leaf-activated` spans (as one such span typically lasts around 6 seconds). These spans would generate 100 samples for each subsystem's time duration, i.e. a histogram / distribution of performance data for each subsystem. Based on that we can ask questions like:

- "What is the average duration for candidate-backing in a relay network with X nodes?"
- "What is the 95th percentile (in terms of duration) for bitfield-signing with X nodes?"

but more importantly, this utility tool will allow us to answer the questions, and be more granular with the experiments we make from a scalability and performance point of view.
